internal/pkg/service/stream/api/mapper/sink_request.go

```diff
@@ -0,0 +1,185 @@
+package mapper
+
+import (
+	"github.com/keboola/go-client/pkg/keboola"
+
+	svcerrors "github.com/keboola/keboola-as-code/internal/pkg/service/common/errors"
+	api "github.com/keboola/keboola-as-code/internal/pkg/service/stream/api/gen/stream"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition/key"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/mapping/table"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/mapping/table/column"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/strhelper"
+)
+
+func (m *Mapper) NewSinkEntity(parent key.SourceKey, payload *api.CreateSinkPayload) (entity definition.Sink, err error) {
+	entity.SourceKey = parent
+
+	// Generate source ID from Name if not set
+	if payload.SinkID == nil || len(*payload.SinkID) == 0 {
+		entity.SinkID = key.SinkID(strhelper.NormalizeName(payload.Name))
+	} else {
+		entity.SinkID = key.SinkID(strhelper.NormalizeName(string(*payload.SinkID)))
+	}
+
+	// Name
+	entity.Name = payload.Name
+
+	// Description is optional
+	if payload.Description != nil {
+		entity.Description = *payload.Description
+	}
+
+	// Sink type
+	entity.Type = payload.Type
+	switch entity.Type {
+	case definition.SinkTypeTable:
+		if tableEntity, err := m.newTableSinkEntity(payload); err == nil {
+			entity.Table = &tableEntity
+		} else {
+			return definition.Sink{}, err
+		}
+	default:
+		return definition.Sink{}, svcerrors.NewBadRequestError(errors.Errorf(`unexpected "type" "%s"`, payload.Type.String()))
+	}
+
+	return entity, nil
+}
+
+func (m *Mapper) UpdateSinkEntity(entity definition.Sink, payload *api.UpdateSinkPayload) (definition.Sink, error) {
+	// Name
+	if payload.Name != nil {
+		entity.Name = *payload.Name
+	}
+
+	// Description
+	if payload.Description != nil {
+		entity.Description = *payload.Description
+	}
+
+	// Type
+	if payload.Type != nil {
+		entity.Type = *payload.Type
+	}
+
+	// Type specific updates
+	switch entity.Type {
+	case definition.SinkTypeTable:
+		if entity.Table == nil {
+			entity.Table = &definition.TableSink{}
+		}
+		if payload.Table != nil {
+			if err := m.updateTableSinkEntity(entity.Table, payload); err != nil {
```

Comment:
Cannot be check of  `payload.Table` inside `updateTableSinkEntity`? The `entity.Table` is set before but we do not push just the `payload.Table` but whole `payload`. So we could do the check in the function where we will return `nil` error and reduce nest of code.